src/main/java/org/folio/qm/util/TenantUtils.java

```diff
@@ -0,0 +1,24 @@
+package org.folio.qm.util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+
+import lombok.experimental.UtilityClass;
+
+import org.folio.spring.DefaultFolioExecutionContext;
+import org.folio.spring.FolioExecutionContext;
+
+@UtilityClass
+public class TenantUtils {
+
+  public static FolioExecutionContext getFolioExecutionContextCopyForTenant(FolioExecutionContext folioExecutionContext, String tenant) {
+    var headers = folioExecutionContext.getAllHeaders() != null
+      ? folioExecutionContext.getAllHeaders()
+      : new HashMap<String, Collection<String>>();
+    headers.put("x-okapi-tenant", Collections.singletonList(tenant));
```

Comment:
```suggestion
    headers.put(XOkapiHeaders.TENANT, Collections.singletonList(tenant));
```
You can use this constant instead.